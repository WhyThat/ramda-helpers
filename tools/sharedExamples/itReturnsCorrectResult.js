export default (result) => {
  test('returns correct result', () => {
    expect(result).toMatchSnapshot();
  });
};
