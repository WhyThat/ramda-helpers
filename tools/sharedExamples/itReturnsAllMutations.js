import itReturnsCorrectResult from './itReturnsCorrectResult';

export default (fn, mutations) => {
  mutations.forEach((mutation, index) => {
    const name = mutation.name || `mutations ${index}`;

    describe(name, () => {
      itReturnsCorrectResult(fn(mutation.props));
    });
  });
};
