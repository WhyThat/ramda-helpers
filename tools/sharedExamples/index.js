
import itReturnsAllMutations from './itReturnsAllMutations';
import itReturnsCorrectResult from './itReturnsCorrectResult';

export {
  itReturnsAllMutations,
  itReturnsCorrectResult,
};
