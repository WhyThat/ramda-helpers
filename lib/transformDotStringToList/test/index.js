import transformToList from '../';
import { itReturnsCorrectResult } from '../../test/sharedExamples';

describe('TransformDotStringToList', () => {
  const mutations = [
    { name: 'With an empty string', props: '' },
    { name: 'With an empty array', props: [] },
    { name: 'With dotted string', props: 'a.b.c' },
    { name: 'With array', props: ['a', 'b', 'c'] },
  ];

  mutations.forEach((mutation) => {
    describe(mutation.name, () => {
      itReturnsCorrectResult(transformToList(mutation.props));
    });
  });
});

