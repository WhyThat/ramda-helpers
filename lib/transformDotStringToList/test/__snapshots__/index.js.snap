// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransformDotStringToList With an empty array returns correct result 1`] = `Array []`;

exports[`TransformDotStringToList With an empty string returns correct result 1`] = `
Array [
  "",
]
`;

exports[`TransformDotStringToList With array returns correct result 1`] = `
Array [
  "a",
  "b",
  "c",
]
`;

exports[`TransformDotStringToList With dotted string returns correct result 1`] = `
Array [
  "a",
  "b",
  "c",
]
`;
